src/components/TextEditor/TextEditor.js

```diff
@@ -1,76 +1,63 @@
-import React, {Fragment} from 'react';
+import React, {Fragment, useCallback, useState, useMemo} from 'react';
 
 import styles from './TextEditor.less';
 
-export default class TextEditor extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isEditing: false,
-            text: props.text
-        }
-    }
-
-    handleChangeText = (event) => {
-        this.setState({
-            text: event.target.value
-        });
-    }
+export default function TextEditor(props) {
+ 
+    const {text, onChangeText, asterisk, placeholder = 'Please enter...'} = props;
 
-    handleChangeButton = () => {
-        this.setState({
-            isEditing: true
-        });
-    }
+    const [isEditing, setEditing] = useState(false);
+    const [textEdit, setText] = useState(text);
 
-    handleCancelButton = () => {
-        this.setState({
-            isEditing: false,
-            text: this.props.text
-        });
-    }
+    const handleChangeText = useCallback(({target}) => {
+        setText(target.value);
+    }, [setText]);
 
-    handleSaveButton = () => {
-        const {onChangeText} = this.props;
-        const {text} = this.state;
+    const handleChangeButton = useCallback(() => {
+        setEditing(true);
+    }, [setEditing]);
+    
+    const handleCancelButton = useCallback(() => {
+        setText(text);        
+        setEditing(false);
+    }, [setText, text, setEditing]);
 
-        onChangeText(text);
+    const handleSaveButton = useCallback(() => {
+        onChangeText(textEdit);
 
-        this.setState({
-            isEditing: false
-        });
-    }
-
-    render() {
-        const {isEditing, text} = this.state;
-        let {asterisk, placeholder} = this.props;
+        setEditing(false);
+    }, [onChangeText, setEditing, textEdit]);
         
-        let showText = ((text === '') ? placeholder : text); 
-        showText = asterisk ? showText.replace(/./g, "*") : showText;
-        showText = (text === '') 
-            ? <div class='text-placeholder'>
-                {showText}
+    const showText = useMemo(() => {
+        if (!text) {
+            return placeholder;
+        }
+        
+        return asterisk ? text.replace(/./g, "*") : text;         
+        
+    }, [text, placeholder, asterisk]);
+
+    return (
+        <div className={styles.flexRow}>
+            <div className={styles.textEditor}>
+                {isEditing
+                    ? (<textarea onChange={handleChangeText}>
+                        {text}
+                    </textarea>)
+                    : (<div className={!text && styles.textPlaceholder}>
+                        {showText}
+                    </div>)
```

Comment:
you messed with brackets :(
you need brackets to start code block and because of brackets you shift `div` or whatever tag on the next line which makes your markup to straight in the column (the same amount of spaces/tabs from line start)